zcash_client_sqlite/src/lib.rs

```diff
@@ -247,13 +251,124 @@ impl fmt::Display for ReceivedNoteId {
 pub struct UtxoId(pub i64);
 
 /// A newtype wrapper for sqlite primary key values for the transactions table.
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 struct TxRef(pub i64);
 
-/// A wrapper for the SQLite connection to the wallet database.
-pub struct WalletDb<C, P> {
+/// A newtype wrapper for sqlite primary key values for the addresses table.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+struct AddressRef(pub(crate) i64);
+
+/// A data structure that can be used to configure custom gap limits for use in transparent address
+/// rotation.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[cfg(feature = "transparent-inputs")]
+pub struct GapLimits {
+    external: u32,
+    internal: u32,
+    ephemeral: u32,
+}
+
+#[cfg(feature = "transparent-inputs")]
+impl GapLimits {
+    /// Constructs a new `GapLimits` value from its constituent parts.
+    ///
+    /// The following gap limits are recommended for use with this crate, and are supplied by the
+    /// [`Default`] implementation for this type.
+    /// - external addresses: 10
+    /// - transparent internal (change) addresses: 5
+    /// - ephemeral addresses: 3
```

Comment:
```suggestion
    /// The gap limits recommended for use with this crate are supplied by the [`Default`]
    /// implementation for this type.
```
(avoid duplication and fix the value for ephemeral addresses which had not been updated).